### PR TITLE
 remove tojs calls for some source methods

### PR DESCRIPTION
**Pull request by codehag**
its a beast.

### CODE REVIEW

File: src/actions/sources/loadSourceText.js [hunk 1]
```diff
@@ -9,20 +9,21 @@ import { setSymbols } from "../ast";
...
+  if (sourceMaps.isOriginalId(id)) {
+    return await sourceMaps.getOriginalSourceText(source.toJS());
   }
 
-  const response = await client.sourceContents(source.id);
+  const response = await client.sourceContents(id);
 
   return {
-    id: source.id,
+    id: id,
```
Comment (jasonLaster) [inline src/actions/sources/loadSourceText.js]: `id, `

File: src/actions/sources/tests/loadSourceText.spec.js [hunk 2]
```diff
@@ -17,20 +17,21 @@ describe("loadSourceText", async () => {
           resolve = r;
         })
     });
-    let source = makeSource("foo", { loadedState: "unloaded" });
+    const id = "foo";
```
Comment (jasonLaster) [inline src/actions/sources/tests/loadSourceText.spec.js]: lol

File: src/test/tests-setup.js [hunk 3]
```diff
@@ -29,5 +29,4 @@ setConfig(config);
 
 process.on("unhandledRejection", (reason, p) => {
   console.log("Unhandled Rejection at:", p, "reason:", reason);
-  throw reason;
```
Comment (jasonLaster) [inline src/test/tests-setup.js]: i think we want the tests to fail :/

File: src/utils/location.js [hunk 4]
```diff
@@ -15,8 +15,8 @@ export function createLocation({
 }: IncompleteLocation): Location {
   return {
     sourceId,
-    line: line || 0,
-    column: column || 0,
+    line: line,
+    column: column,
```
Comment (jasonLaster) [inline src/utils/location.js]: line,
column,

Comment (jasonLaster) [reply src/utils/location.js]: but yeah. nice!

File: src/workers/search/project-search.js [hunk 5]
```diff
@@ -3,11 +3,10 @@
  * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
 
 // Maybe reuse file search's functions?
-import { isLoaded } from "../../utils/source";
 
 export function findSourceMatches(source, queryText) {
-  const { text } = source;
-  if (!isLoaded(source) || !text || queryText == "") {
+  const text = source.text;
+  if (source.loadedState !== "loaded" || !text || queryText == "") {
```
Comment (jasonLaster) [inline src/workers/search/project-search.js]: haha - yep. This *does* make it harder to use our utils now that expect SourceRecord :)

but when we go to I Records, it'll be transparent 😉 and we can use the utils again

File: src/workers/search/tests/project-search.spec.js [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import { findSourceMatches } from "../project-search";
+import I from "immutable";
```
Comment (jasonLaster) [inline src/workers/search/tests/project-search.spec.js]: do we need this?

File: src/test/tests-setup.js (hunk 3 above)
Comment (codehag) [reply src/test/tests-setup.js]: the test fails anyway, but this exits the test runner, making it hard to see where the test failed

File: src/workers/search/project-search.js (hunk 5 above)
Comment (codehag) [reply src/workers/search/project-search.js]: another way to think of this, is maybe we insulate our workers more, so they are more independant

File: src/workers/search/tests/project-search.spec.js (hunk 6 above)
Comment (codehag) [reply src/workers/search/tests/project-search.spec.js]: we can just get map

**Comment by codehag** [comment]
talos dif

    Averages:
  
    simple.jsdebugger.open.DAMP: 602.802ms (0.00% difference BASECOMMIT)
    simple.jsdebugger.reload.DAMP: 136.095ms (0.00% difference BASECOMMIT)
    simple.jsdebugger.close.DAMP: 90.638ms (0.00% difference BASECOMMIT)
    complicated.jsdebugger.open.DAMP: 972.618ms (0.00% difference BASECOMMIT)
    complicated.jsdebugger.reload.DAMP: 1367.705ms (0.00% difference BASECOMMIT)
    complicated.jsdebugger.close.DAMP: 729.817ms (0.00% difference BASECOMMIT)
    
    Commit Message: remove tojs calls for some source methods
    Link: https://www.github.com/devtools-html/debugger.html/commit/0f788a658193aeaad1b498e32f2355975acca18f
    Author: codehag <yulia.startsev@gmail.com>
    Date:   Fri Dec 8 17:54:28 2017 +0100

    Averages:
  
    simple.jsdebugger.open.DAMP: 530.92ms (-13.54% difference)
    simple.jsdebugger.reload.DAMP: 111ms (-22.61% difference)
    simple.jsdebugger.close.DAMP: 67.242ms (-34.79% difference)
    complicated.jsdebugger.open.DAMP: 824.508ms (-17.96% difference)
    complicated.jsdebugger.reload.DAMP: 1180.155ms (-15.89% difference)
    complicated.jsdebugger.close.DAMP: 625.727ms (-16.64% difference)

File: src/actions/sources/loadSourceText.js [hunk 7]
```diff
@@ -51,32 +54,34 @@ export function loadSourceText(source: Source) {
...
+    if (isLoading(source) || requests.has(id)) {
+      return requests.get(id);
     }
 
-    requests.set(source.id, deferred.promise);
+    requests.set(id, deferred.promise);
     await dispatch({
       type: "LOAD_SOURCE_TEXT",
-      source: source,
+      source: { id: source.get("id") },
```
Comment (jasonLaster) [inline src/actions/sources/loadSourceText.js]: lets change this to `sourceId`

File: src/test/tests-setup.js (hunk 3 above)
Comment (jasonLaster) [reply src/test/tests-setup.js]: i see. okay

**Pull request merged**